bin/test_runner.rb

```diff
@@ -286,10 +292,9 @@ def check_todo_list
   def run
     ####  main running loop
     while true
```

Comment:
I still feel this is analogous of using `cat /file/name | awk '/findthis/'` instead of `grep findthis /file/name`.  Strongly suggest using `loop do` - as other online documentation also suggests.